plugins/ingestion-kafka/src/test/java/org/opensearch/plugin/kafka/KafkaUtils.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.kafka;
+
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.AdminClientConfig;
+import org.apache.kafka.clients.admin.KafkaAdminClient;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.kafka.clients.admin.TopicDescription;
+import org.apache.kafka.common.KafkaFuture;
+import org.apache.kafka.common.errors.TopicExistsException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.Assert;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+import org.testcontainers.shaded.com.google.common.collect.ImmutableMap;
+
+import static org.testcontainers.shaded.org.awaitility.Awaitility.await;
+
+public class KafkaUtils {
+    private static final Logger LOGGER = LogManager.getLogger(KafkaUtils.class);
+
+    /**
+     * Creates kafka topic
+     *
+     * @param topicName the topic name
+     * @param bootstrapServer kafka bootstrap server list
+     */
+    public static void createTopic(String topicName, String bootstrapServer) {
+        createTopic(topicName, 1, bootstrapServer);
+    }
+
+    public static void createTopic(String topicName, int numOfPartitions, String bootstrapServers) {
+        try {
+            getAdminClient(bootstrapServers, (client -> {
+                NewTopic newTopic = new NewTopic(topicName, numOfPartitions, (short) 1);
+                client.createTopics(List.of(newTopic));
+                return true;
+            }));
+
+        } catch (TopicExistsException e) {
+            // Catch TopicExistsException otherwise it will break maven-surefire-plugin
+            LOGGER.warn("Topic {} already existed", topicName);
+        }
+
+        // validates topic is created
+        await().atMost(3, TimeUnit.SECONDS).untilAsserted(() -> { Assert.assertTrue(checkTopicExistence(topicName, bootstrapServers)); });
```

Comment:
I haven't used this awaitability library before, but can you simplify this to avoid the asserts?

```suggestion
        await().atMost(3, TimeUnit.SECONDS).until(() -> checkTopicExistence(topicName, bootstrapServers));
```

Note that we do have this [assertBusy() method](https://github.com/opensearch-project/OpenSearch/blob/33be5a9d161e496517842ded0bc4b04ecca674ff/test/framework/src/main/java/org/opensearch/test/OpenSearchTestCase.java#L1108) and variants in the base test class for waiting for async conditions. I'm not opposed to adding a new library if it is helpful, but is there a specific reason to prefer the new library versus the existing utilities?

Reply:
https://github.com/awaitility/awaitility is a popular and powerful lib for async testing, and it's kind of the defacto lib for this purpose. so I highly suggest we consider it

Reply:
and yes, good suggestion to use `until` directly